pkg/endpoint/endpoint.go

```diff
@@ -489,9 +489,7 @@ func CreateHostEndpoint(owner regeneration.Owner, proxy EndpointProxy, allocator
 	ep := createEndpoint(owner, proxy, allocator, 0, ifName)
 	ep.isHost = true
 	ep.nodeMAC = mac
-	ep.DatapathConfiguration = models.EndpointDatapathConfiguration{
-		RequireEgressProg: true,
-	}
+	ep.DatapathConfiguration = NewDatapathConfiguration()
```

Comment:
Previously this hardcoded `RequireEgressProg: true` for the host, but with the new option this will only be the case when `config.EnableEndpointRoutes` is true. Do we need to take a second look at host endpoint + no endpoint routes?

Reply:
Nice catch! I checked that case already, but should have added a note in the commit.

The host endpoint actually ignores `RequireEgressProg` and attaches to both ingress and egress (cf. `reloadHostDatapath()`). I'm setting it here just for consistency with other endpoints, but you could argue it should be true just to be consistent with the datapath state. I don't have a strong opinion either way.

Reply:
In that case I'm not too worried about it, one could argue that the `RequireEgressProg` semantics is specific to managed endpoints rather than hosts.